contrib/plugins/qt-creator-project.py

```diff
@@ -0,0 +1,231 @@
+                    elif 'plugin:buildstep' in line:
+                        addBuildSteps(sharedFile, 'dev' if devMode else 'build', project, destination)
+                    else:
+                        sharedFile.write(line)
+
+def projectGenerator(argv, bobRoot):
+    parser = argparse.ArgumentParser(prog="bob project", description='Generate Project Files')
+    parser.add_argument('package', type=str, nargs='?',
+                        help="Sub-package to start listing from")
+    devBuild = parser.add_mutually_exclusive_group()
+    devBuild.add_argument('-d', '--dev', default=True, action='store_true',
+                        help="Use bob dev mode in project files")
+    devBuild.add_argument('-b', '--build', default=False, action='store_false', dest='dev',
+                        help="Use bob build in project files")
+    parser.add_argument('-u', '--update', default=False, action='store_false',
+                        help="Update project files (.files, .includes)")
+    parser.add_argument('--destination', metavar="DEST",
+        help="Destination of project files")
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--sandbox', action='store_true', default=True,
+        help="Enable sandboxing")
+    group.add_argument('--no-sandbox', action='store_false', dest='sandbox',
+        help="Disable sandboxing")
+    args = parser.parse_args(argv)
+
+    recipes = RecipeSet()
+    recipes.defineHook('releaseNameFormatter', LocalBuilder.releaseNameFormatter)
+    recipes.defineHook('developNameFormatter', LocalBuilder.developNameFormatter)
+    recipes.defineHook('developNamePersister', LocalBuilder.developNamePersister)
+    recipes.parse()
+
+    if args.dev:
+        nameFormatter = recipes.getHook('developNameFormatter')
+        developPersister = recipes.getHook('developNamePersister')
+        nameFormatter = developPersister(nameFormatter)
+    else:
+        nameFormatter = recipes.getHook('releaseNameFormatter')
+        nameFormatter = LocalBuilder.releaseNamePersister(nameFormatter)
+    nameFormatter = LocalBuilder.makeRunnable(nameFormatter)
+
+    roots = recipes.generatePackages(nameFormatter, sandboxEnabled=args.sandbox).values()
+    if args.package:
+        steps = [ s for s in args.package.split("/") if s != "" ]
+        for s in steps:
+            roots = [ r for r in roots if r.getName() == s ]
+            if not roots:
+                print("{}: not found in '{}'".format(s, "/".join(stack)),
+                      file=sys.stderr)
+                sys.exit(1)
+            self = roots[0]
+            roots = [ d.getPackage() for d in roots[0].getAllDepSteps() ]
+            roots.append(self)
+    else:
+       print("Error: No package given")
+       return -1
+    generateQtProject(roots, args.dev, args.destination, args.package, args.update)
+
+
+
+manifest = {
+    'apiVersion' : "0.1",
+    'hooks' : {
+        'qt-project' : projectGenerator,
+    }
```

Comment:
:+1:  but I'd like to use `generators` instead of `projectGenerators`, thinking of a generator to generate tags of your actual used sources or something like this.